src/bot/userInfoModal.ts

```diff
@@ -0,0 +1,95 @@
+ * @param {ModalData} data - The modal data
+ * @returns {Promise<ModalBuilder>} - The created modal
+ */
+const createModal = async (data: BotConfig['modal']): Promise<ModalBuilder> => {
+  const modal = new ModalBuilder().setCustomId('registerModal').setTitle(data.title);
+
+  const firstNameComponent = new ActionRowBuilder().addComponents(new TextInputBuilder()
+    .setCustomId('firstName')
+    .setLabel(data.firstNameLabel)
+    .setStyle(TextInputStyle.Short)
+    .setMinLength(1)
+    .setMaxLength(16)
+    .setPlaceholder(data.firstNamePlaceholder)
+    .setRequired(true)) as ActionRowBuilder<TextInputBuilder>;
+  const nameComponent = new ActionRowBuilder().addComponents(new TextInputBuilder()
+    .setCustomId('name')
+    .setLabel(data.familyNameLabel)
+    .setStyle(TextInputStyle.Short)
+    .setPlaceholder(data.familyNamePlaceholder)
+    .setRequired(true)) as ActionRowBuilder<TextInputBuilder>;
+  const graduationYearComponent = new ActionRowBuilder().addComponents(new TextInputBuilder()
+    .setCustomId('graduationYear')
+    .setLabel(data.graduationYearLabel)
+    .setStyle(TextInputStyle.Short)
+    .setMinLength(4)
+    .setMaxLength(4)
+    .setPlaceholder(data.graduationYearPlaceholder)
+    .setRequired(true)) as ActionRowBuilder<TextInputBuilder>;
+  const educationComponent = new ActionRowBuilder().addComponents(new TextInputBuilder()
+    .setCustomId('education')
+    .setLabel(data.educationLabel)
+    .setStyle(TextInputStyle.Short)
+    .setMinLength(2)
+    .setMaxLength(5)
+    .setPlaceholder(data.educationPlaceholder)
+    .setRequired(true)) as ActionRowBuilder<TextInputBuilder>;
+
+  modal.addComponents([
+    firstNameComponent,
+    nameComponent,
+    graduationYearComponent,
+    educationComponent,
+  ]);
+  return modal;
+};
+
+/**
+ * Validate the modal
+ * @param {ModalSubmitInteraction} interaction - Interaction data
+ * @param {string} roleId - The role to add to the user
+ * @return {void}
+ */
+async function onModalSubmit(interaction: ModalSubmitInteraction, roleId: string): Promise<void> {
+  if (!interaction.member) {
+    return;
+  }
+
+  const firstName = interaction.fields.getTextInputValue('firstName');
+  const familyName = interaction.fields.getTextInputValue('name');
+  const graduationYear = interaction.fields.getTextInputValue('graduationYear');
+  const education = interaction.fields.getTextInputValue('education');
+
+  const separatorLength = 5; // spaces and dash
+  const suffixLength = graduationYear.length + education.length;
+  const fullNameLength = firstName.length + familyName.length;
+  if (separatorLength + suffixLength + fullNameLength > 32) {
+    await (interaction.member as GuildMember).setNickname(`${firstName} ${familyName.charAt(0).toUpperCase()}. - ${graduationYear} ${education}`);
+  } else {
+    await (interaction.member as GuildMember).setNickname(`${firstName} ${familyName.toUpperCase()} - ${graduationYear} ${education}`);
```

Comment:
I think we should rethink this part:
- Education field does not make sense when people do something completely different (and it complexifies the logic with double degree or doctorate)
- Maybe we should just keep the first name and the last name, troncaturing the string to keep the graduation year at the end